---
fig-cap-location: top
---

# **Day 1 learning objectives**

Today, we will introduce you to the first steps of analysing RNA sequencing (RNA-seq) data using the command line. We will learn about the pre-processing workflow we use to convert raw sequence reads to analysis-ready count data. 

### **Introductory slides**
- Understand how RNA-seq data is generated
- Review the applications of RNA-seq
- Understand experimental design considerations for differential expression (DE) analysis
- Understand the basic RNA-seq DE analysis workflow

### **Run the nf-core/rnaseq pipeline**
- Set up your computer for this workshop series
- Log in to your Nimbus instance
- Download the input data files 
- Run the nf-core/rnaseq command to excecute the pipeline 

### **Why use nf-core workflows?**
- Understand why Nextflow and nf-core are good options for reproducible and portable bioinformatics workflows

### **The ins and outs of nf-core-rnaseq**
- Write and run a basic nf-core/rnaseq command to perform RNA-seq data pre-processing
- Understand what input files and parameters are required to run the command
- Review the output files generated by the nf-core/rnaseq pipeline 
- Understand how to adjust the run command to customise the workflow 

### **The RNAseq pre-processing workflow**
- Understand the different steps in a typical RNA-seq pre-processing pipeline
- Evaluate the results and outputs generated by the nf-core/rnaseq pipeline 
- Understand how to perform quality checking of raw sequence data 
- Understand the requirements for trimming raw RNA-seq reads 
- Understand the process of aligning RNA-seq reads to a reference genome 
- Understand how read alignments are converted to gene count data 
